src/App.tsx

```diff
@@ -8,53 +8,135 @@ import { PostsList } from './components/PostsList';
+import { getPosts, getUsers } from './api/api';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [errorNoPosts, setErrorNoPosts] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const getUserPosts = useCallback(async (): Promise<void> => {
+    if (!selectedUser) {
+      return;
+    }
 
-              <Loader />
+    try {
+      setLoading(true);
+      setError(false);
+      setErrorNoPosts(false);
+      setPosts([]);
+      const dataPosts = await getPosts(selectedUser.id);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+      if (dataPosts.length === 0) {
+        setErrorNoPosts(true);
+      } else {
+        setErrorNoPosts(false);
+      }
+
+      setPosts(dataPosts as Post[]);
+    } catch {
+      setError(true);
+    } finally {
+      setLoading(false);
+    }
+  }, [selectedUser]);
+
+  useEffect(() => {
+    if (selectedUser !== null) {
+      setSelectedPost(null);
+      getUserPosts();
+    }
+  }, [selectedUser, getUserPosts]);
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      try {
+        const fetchedUsers = await getUsers();
+
+        setUsers(fetchedUsers);
+      } catch {}
+    };
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+    fetchUsers();
+  }, []);
+
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                />
               </div>
 
-              <PostsList />
+              <div className="block" data-cy="MainContent">
+                {selectedUser === null ? (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                ) : null}
+                {loading ? <Loader /> : null}
```

Comment:
Don't need to use `null`, fix it everywhere
```suggestion
                {loading && <Loader />}
```